linode/networkingip/framework_resource.go

```diff
@@ -27,9 +30,52 @@ type Resource struct {
 	helper.BaseResource
 }
 
+func (r *Resource) ModifyPlan(
+	ctx context.Context,
+	req resource.ModifyPlanRequest,
+	resp *resource.ModifyPlanResponse,
+) {
+	// reconcileIPReassignment marks changes to linode_id as RequiresReplace
+	// if the operation cannot be completed using a resource update.
+	reconcileIPReassignment := func() {
```

Comment:
I used an anonymous func pattern here to keep it modular and to let us make use of guard clauses 🙂